assets/app/scripts/controllers/newfromtemplate.js

```diff
@@ -0,0 +1,161 @@
+          };
+          // TODO: Determine how to generate Help Links - would they be annotations on the template?
+          var helpLinks = [ { title:"Additional Information", link: "https://github.com/openshift/origin" } ];
+          TaskList.add(titles, helpLinks, function() {
+            var d = $q.defer()
+            DataService.createList(config.items, $scope).then(
+              function(result) {
+                var alerts = [];
+                var hasErrors = false;
+                if (result.failure.length > 0) {
+                  result.failure.forEach(
+                    function(failure) {
+                      var objectName = "";
+                      if (failure.data && failure.data.details) {
+                        objectName = failure.data.details.kind + " " + failure.data.details.id;
+                      }
+                      alerts.push({ type: "error", message: "Cannot create " + objectName, details: failure.data.message })
+                      hasErrors = true;
+                    }
+                  );
+                } else {
+                  alerts.push({ type: "success", message: "All items in template " + $scope.template.metadata.name +
+                    " were created successfully."});
+                }
+                d.resolve({alerts: alerts, hasErrors: hasErrors});
+              }
+            );
+            return d.promise;
+          });
+          $location.path("/project/" + $scope.projectName + "/overview");
+        },
+        function(result) { // failure
+          $scope.alerts = [
+            { 
+              type: "error", 
+              message: "An error occurred processing the template.",
+              details: "Status: " + result.status + ". " + result.data,
+            }
+          ]
+        }
+      );
+    }
+    
+    $scope.toggleOptionsExpanded = function() {
+      $scope.optionsExpanded = !$scope.optionsExpanded
+    }
+    
+    $scope.paramPlaceholder = function(param) {
+      if (param.generate) {
+        return "(generated)"
+      } else {
+        return ""
+      }
+    }
+    
+    $scope.paramValue = function(param) {
+      if (!param.value && param.generate) {
+        return "(generated)"
+      } else {
+        return param.value
+      }
+    }
+    
+    
+    var name = $routeParams.name;
+    var namespace = $routeParams.namespace;
+    var url = $routeParams.url;
+    
+    if (!name && !url) {
+      errorPage("Cannot create from template: a template name or URL was not specified.");
+      return;
+    }
+
+    $scope.templateUrl = url;
+    $scope.emptyMessage = "Loading..."; 
+    $scope.alerts = [];
+    $scope.projectPromise = $.Deferred();
+    $scope.projectName = $routeParams.project
+    $scope.projectPromise.resolve({ metadata: { name: namespace || $scope.projectName }});
+
+    DataService.get("templates", name, $scope).then(
```

Comment:
@jwforres - Is this a change in a different PR? (that you can pass {namespace:ns} instead of context). I had made a change to the get/delete/create methods that took a namespace in the opts, since context could still be used for other stuff. I still have not added the change to this PR.